packages/orchestration/src/exos/time.js

```diff
@@ -0,0 +1,76 @@
+import { RelativeTimeRecordShape, TimeMath } from '@agoric/time';
+import { VowShape } from '@agoric/vow';
+import { watch, allVows } from '@agoric/vow/vat.js';
+import { makeHeapZone } from '@agoric/zone';
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+
+/**
+ * @import {Remote} from '@agoric/internal';*
+ * @import {RelativeTimeRecord, TimerBrand, TimerService, TimestampRecord} from '@agoric/time';
+ * @import {Vow} from '@agoric/vow';
+ * @import {Zone} from '@agoric/zone';
+ */
+
+export const SECONDS_PER_MINUTE = 60n;
+export const NANOSECONDS_PER_SECOND = 1_000_000_000n;
+
+/**
+ * @param {Remote<TimerService>} timerService
+ * @param {Zone} [zone]
+ */
+
+export const makeTimestampHelper = (timerService, zone = makeHeapZone()) => {
+  /** @type {TimerBrand | undefined} */
+  let brandCache;
+  const getBrand = () => {
+    if (brandCache) return brandCache;
+    return watch(E(timerService).getTimerBrand(), {
+      onFulfilled: timerBrand => {
+        return timerBrand;
+      },
```

Comment:
`brandCache` is never set. A quick fix would be,
```suggestion
      onFulfilled: timerBrand => {
        brandCache = timerBrand;
        return timerBrand;
      },
```

I think we should have a _provide_ helper for vows akin to `provideSingleton` for promises but that can be a separate refactor.

Reply:
🤦‍♂️

Reply:
I see the fix but I'll leave this thread open to suggest leaving an XXX comment for creating a helper for lazily fetching a value for vows

Reply:
I'm not sure I follow this suggestion completely. Are you talking about caching the result of the request, or lazily fetching?  It seems maybe the former for the current revision? Is there a ticket?

It would be cool if we could init state lazily for an exo - sort of like the `defer` behavior for web script tags . The exo state initialization can include remote calls that can be deferred outside the first crank. It wouldn't be interactive or  accept messages until `initState` has completed.

Reply:
> caching the result of the request, or lazily fetching

Both. That's what this is doing now. I'm suggesting a comment that says `XXX this is a common use case that should have a helper`.

To clarify, I have in mind something like this we already have:
```
      getBrand() {
        return provideSingleton(this.state, 'brandCache', E(timerService).getTimerBrand());
      },
```
but that would return a vow.


There is no ticket because this is not a release requirement. Just something to consider improving as we iterate on the code.